File: app/src/test/java/com/task/ui/component/news/TestModelsGenerator.java
package com.task.ui.component.news;

import com.task.data.remote.ServiceResponse;
import com.task.data.remote.dto.NewsItem;
import com.task.data.remote.dto.NewsModel;

import java.util.ArrayList;
import java.util.List;

import static com.task.data.remote.ServiceError.ERROR_CODE;
import static com.task.data.remote.ServiceError.SUCCESS_CODE;

/**
 * Created by ahmedeltaher on 3/8/17.
 */

public class TestModelsGenerator {

    public NewsModel generateNewsModel(String stup) {
        NewsModel newsModel = new NewsModel();
        newsModel.setCopyright(stup);
        newsModel.setLastUpdated(stup);
        newsModel.setSection(stup);
        newsModel.setStatus(stup);
        newsModel.setNumResults(25L);
        List<NewsItem> newsItems = new ArrayList<>();
        for (int i = 0; i < 25; i++) {
            newsItems.add(generateNewsItemModel(stup));
        }
        newsModel.setNewsItems(newsItems);
        return newsModel;
    }

    public NewsItem generateNewsItemModel(String stup) {
        NewsItem newsItem = new NewsItem();
        newsItem.setTitle(stup);
        newsItem.setAbstract(stup);
        newsItem.setUrl(stup);
        return newsItem;
    }

    public ServiceResponse getNewsSuccessfulModel() {
        String stupString = "this is temp string";
        NewsModel newsModel = generateNewsModel(stupString);
        return new ServiceResponse(SUCCESS_CODE, newsModel);
    }

    public ServiceResponse getNewsErrorModel() {
        return new ServiceResponse(ERROR_CODE, null);
    }
}

File: app/src/test/java/com/task/ui/component/news/TestModelsGenerator.kt
package com.task.ui.component.news

import com.google.gson.Gson
import com.task.data.remote.dto.NewsItem
import com.task.data.remote.dto.NewsModel
import java.io.File
import java.util.*

/**
 * Created by ahmedEltaher on 3/8/17.
 */

class TestModelsGenerator {
    private var newsModel: NewsModel

    init {
        val gson = Gson()
        val jsonString = getJson("NewsApiResponse.json")
        newsModel = gson.fromJson(jsonString, NewsModel::class.java)
    }

    fun generateNewsModel(): NewsModel {
        return newsModel
    }

    fun generateNewsModelWithEmptyList(): NewsModel {
        newsModel.newsItems = ArrayList()
        return newsModel
    }

    fun generateNewsItemModel(): NewsItem {
        return newsModel.newsItems[0]
    }

    fun getStupSearchTitle(): String {
        return newsModel.newsItems[0].title
    }


    /**
     * Helper function which will load JSON from
     * the path specified
     *
     * @param path : Path of JSON file
     * @return json : JSON from file at given path
     */

    private fun getJson(path: String): String {
        // Load the JSON response
        val uri = this.javaClass.classLoader?.getResource(path)
        val file = File(uri?.path)
        return String(file.readBytes())
    }
}
